test/e2e/e2e.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/reporters"
+	"github.com/onsi/gomega"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	gcecloud "k8s.io/kubernetes/pkg/cloudprovider/providers/gce"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+const (
+	// podStartupTimeout is the time to allow all pods in the cluster to become
+	// running and ready before any e2e tests run. It includes pulling all of
+	// the pods (as of 5/18/15 this is 8 pods).
+	podStartupTimeout = 10 * time.Minute
+)
+
+var (
+	cloudConfig = &testContext.CloudConfig
+
+	reportDir = flag.String("report-dir", "", "Path to the directory where the JUnit XML reports should be saved. Default is empty, which doesn't generate these reports.")
+)
+
+type E2ETests struct {
+}
+
+func NewE2ETests() *E2ETests {
+	tests := &E2ETests{}
+	tests.RegisterE2EFlags()
+	return tests
+}
+
+// Sanity checks that the e2e's have the basic information required to run before launching them.
+func (e2e *E2ETests) IsSetup() bool {
```

Comment:
@ixdy in the follow on Jira where we improve testContext, I'm thinking all the config checking can be externalized. Again the goal here is to minimally modify the e2e so that if they are misconfigured by an extending component, there is a fast failure message .   

Context: The likelihood of such misconfiguration happening is a little higher now, since the e2e'go no longer uses "init", the ordering of config is not enforced for us by golang, but rather, by the e2e_test.go file...

...So I've just externalized the validation so that it preempts noisy failures that happen when flag bindings weren't  called properly in order .